docs/DeveloperGuide.adoc

```diff
@@ -96,64 +112,413 @@ The `UI` component,
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+==== Contact Logic component
+.Structure of the Contact Logic Component
+image::ContactLogicClassDiagram.png[]
+
+. `Contact Logic` is a sub-component of `Logic`.
+. `TaglineParser` will pass a user input that can be classified as a contact command (i.e. has __"contact "__ prefix),
+to the `ContactCommandParser` without including the __"contact"__ keyword, e.g. `TaglineParser` will only pass
+__"create --n Bob"__ instead of __"contact create --n Bob"__.
+. `ContactCommandParser` identifies the type of contact command and passes the argument string to the respective command
+parser. For example, `ContactCommandParser` will pass __"--n Bob"__ to `CreateContactParser` if it receives
+__"create --n Bob"__ as an input.
+. This results in a `ContactCommand` object which is returned to the `LogicManager`.
+. The command execution can affect the `ContactModel`.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the
+`execute("contact create --n Bob")` API call.
 
-.Interactions Inside the Logic Component for the `delete 1` Command
-image::DeleteSequenceDiagram.png[]
+.Interactions Inside the Logic Component for the `contact create --n Bob` Command
+image::ContactCreateSequenceDiagram.png[]
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+==== Note Logic component
+
+[[fig-NoteLogicClassDiagram]]
+.Structure of the Note Logic Component
+image::NoteLogicClassDiagram.png[]
+
+.  `Note Logic` is a sub-component of `Logic`.
+.  It obtains the user command parsed by `TaglineParser` through the `NoteCommandParser` class.
+.  The user command is passed to the respective command parser. E.g. __"note delete 1"__ will be passed to `DeleteNoteParser`.
+.  This results in a `NoteCommand` object which is returned to the `LogicManager`.
+.  The command execution can affect the `NoteModel` (e.g. adding a note).
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("note delete 1")` API call.
+
+.Interactions Inside the Logic Component for the `note delete 1` Command
+image::NoteDeleteSequenceDiagram.png[]
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
 image::ModelClassDiagram.png[]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
+*API* : link:{repoURL}/src/main/java/tagline/model/Model.java[`Model.java`]
 
 The `Model`,
 
 * stores a `UserPref` object that represents the user's preferences.
+* manages Address Book data through `ContactModel` sub-component.
+* manages Note Book data through `NoteMode` sub-component.
+
+[[Design-ContactModel]]
+==== Contact Model component
+
+.Structure of the Contact Model Component
+image::ContactModelClassDiagram.png[Contact Model Diagram, 625, 500]
```

Comment:
Composition of contact with the various 6 fields does not comply with the uml notation. 
![image](https://user-images.githubusercontent.com/39525563/67546027-c0b22480-f72d-11e9-858a-d414c3b75c82.png)